lib/evmone/eof.cpp

```diff
@@ -112,11 +97,12 @@ std::variant<EOFSectionHeaders, EOFValidationError> validate_eof_headers(bytes_v
                 section_num = static_cast<uint16_t>((section_number_hi << 8) | section_number_lo);
                 if (section_num == 0)
                     return EOFValidationError::zero_section_size;
+                expected_section_id = DATA_SECTION;
                 state = State::section_size;
                 break;
             }
             default:
-                return EOFValidationError::unknown_section_id;
```

Comment:
Was this executable?

Reply:
Yes it was returned when section_id is unknown, but now it fails earlier with new error.